driver/src/main/java/org/neo4j/driver/internal/FailableCursor.java

```diff
@@ -22,5 +22,6 @@
 
 public interface FailableCursor
 {
+    CompletionStage<Throwable> consumeAsync();
     CompletionStage<Throwable> failureAsync();
```

Comment:
Is this still needed? At least from what I read, you using `consumeAsync` now where `failureAsync` on the failable cursor has been used before.

I found that the failable cursor is used as interface for the `RxStatementResultCursor` as well… Hmm, it feels like that area could need some unification as well.

Reply:
The different of these two method:
consumeAsync: Discard all unconsumed records and return if any error for the whole execution and streaming. Used in `StatementRunner#close`, where after this boundary, all records are discarded.
failureAsync: Buffer all unconsumed records into memory and return if any error for the whole execution and streaming. Used for nested queries between `session#run`s, so that the second run will not discard all previous unconsumed run records.